python/mlx/nn/layers/linear.py

```diff
@@ -26,20 +41,92 @@ class Linear(Module):
 
     def __init__(self, input_dims: int, output_dims: int, bias: bool = True):
         super().__init__()
-        scale = math.sqrt(1 / input_dims)
+        self.input_dims = input_dims
+        self.output_dims = output_dims
```

Comment:
You don't need to record these.  You can use `self.weight.shape` below.

Reply:
I thought keeping this variables would make debugging and reading the code easier. I'll be changing it to use `self.weight.shape` in the next commit.

Reply:
Actually I think it's ok to keep it if it's needed for `reset_parameters()`, but let's include it in a follow on PR when we add that.

Reply:
What do you think should be included in the initialization? I think using 'reset_parameters' is important for any solution. It makes it easy to set up weights, especially for those who prefer a simple approach without specific requirements

Reply:
I'm not sure I understand your question. 

I agree we should have a `reset_parameters` (or some equivalent functionality). My only point is that it should be module general rather than specific to the linear layer. That way you could do something like `module.reset_parameters()`

Reply:
For that to work, we probably need an initialize function or something like that to be part of the base Module class? Idk I haven't thought about it much. Might be worth looking at how other frameworks do it

Reply:
I agree with you. I was addressing the case where you want to initialize specific layers while leaving others. We can leave it for future work.

Reply:
Got it. I've reverted the changes for now, and we'll work on that in a future PR.